integration_tests/backend/utils.py

```diff
@@ -0,0 +1,69 @@
+import time
+import uuid
+
+from aqueduct.enums import ExecutionStatus
+
+import aqueduct
+
+
+def wait_for_flow_runs(
+    client: aqueduct.Client, flow_id: str, num_runs: int = 1, expect_success: Optional[bool] = None
+) -> int:
+    """
+    Returns only when the specified flow has run successfully at least `num_runs` times.
+    Any run failure is not tolerated. Will timeout after a few minutes.
+
+    Returns:
+        The number of successful runs this flow has performed.
+    """
+    timeout = 300
+    poll_threshold = 5
+    begin = time.time()
+
+    while True:
+        assert time.time() - begin < timeout, "Timed out waiting for workflow run to complete."
+
+        time.sleep(poll_threshold)
+
+        # A flow has been successfully published if it makes at least one workflow run, and
+        # all its workflow runs have executed successfully.
+        flow = client.flow(flow_id)
+        flow_runs = flow.list_runs()
+        if len(flow_runs) == 0:
+            continue
+
+        statuses = [flow_run["status"] for flow_run in flow_runs]
+
+        # Continue checking as long as there are still runs pending.
+        if any(status == ExecutionStatus.PENDING for status in statuses):
+            continue
+
+        if len(flow_runs) < num_runs:
+            continue
+
+        if expect_success is not None:
```

Comment:
This & `expect_success` default value is the only change from the SDK integration test utils. Only requiring execution of each flow run to not be pending.